### ISSUE TITLE
rustdoc: macros expanded from macros don't get their rules documented correctly

### ISSUE THREAD

**Issue by remexre** [rust-lang/rust#89367]
When writing a `macro_rules!` macro that expands to another `macro_rules!` macro, the Rustdoc shows the source-code tokens pre-outer-macro-expansion in the rules for the inner macro. I was expecting to see something like the expansion in `trace_macros!()`, which does have the post-expansion rules.

### Meta

`rustdoc --version --verbose`:
```
rustdoc 1.53.0
binary: rustdoc
commit-hash: unknown
commit-date: unknown
host: x86_64-unknown-linux-gnu
release: 1.53.0
LLVM version: 12.0.1
```

**Comment by jyn514** [comment]
@remexre can you give an example of this? I don't know what the difference would be between pre- and post-expansion.

**Comment by jyn514** [comment]
> I can make a minimized example tonight if that'd be more helpful.

That would be great, thanks!

**Comment by remexre** [comment]
Okay, the following program:

```rust
macro_rules! foo {
    ($name:ident => $($body:tt)*) => { foo!([$], $name, $($body)*); };
    ([$dollar:tt], $name:ident, $($body:tt)*) => {
        #[macro_export]
        macro_rules! bar {
            ($dollar $name :tt) => { $($body)* };
        }
    };
}

foo!(msg => println!($msg));

fn main() {
    bar!("Hello, world!");
}
```

gets the `bar` macro documented as:

```rust
macro_rules! bar {
    ($dollar $name :tt) => { ... };
}
```

rather than as

```rust
macro_rules! bar {
    ($ msg :tt) => { ... };
}
```

**Comment by camelid** [comment]
This is fixed now, probably by #86282 (cc #45939).

![image](https://user-images.githubusercontent.com/37223377/147501141-7b6f17d2-7ac2-48ee-bba8-d2a119371b20.png)
